brave/brave-core
# feat(wallet): Choose Quote Provider

## Description 

Introduces a `Choose quote provider` modal on the `Swap` and `Bridge` screens


Resolves <https://github.com/brave/brave-browser/issues/39472>



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Switching Quote Providers:
1. Open the `Wallet` and navigate to the `Swap` screen
2. Select some tokens to swap and get a quote
3. Open the `Choose quote provider` modal and change providers
4. It should instantly fetch a quote for from that provider.

https://github.com/brave/brave-core/assets/40611140/f78df8d4-9372-442e-af80-bfe2cf411bfe

## Test Available Quote Providers:
1. Open the `Wallet` and navigate to the `Swap` screen
2. Select an `Ethereum` token to swap
3. Open the `Choose quote provider` modal
4. `Jupiter` should be grayed out and `Not available`
5. Change the `Ethereum` token to a `Solana` token
6. Open the `Choose quote provider` modal
7. `Ox` should be grayed out and `Not available`
8. Navigate to the `Bridge` screen and open the `Choose quote provider` modal
9. `Ox` and `Jupiter` should be grayed out and `Not available`

https://github.com/brave/brave-core/assets/40611140/b05c11bd-54bc-4416-ab8c-ebe1a70d7bbb

## Test Unsupported Provider Error:
1. Open the `Wallet` and navigate to the `Swap` screen
2. Open the `Choose quote provider` modal and select `0x`
3. Now select a `Solana` token to swap
4. You should see the error message to `Change provider`

https://github.com/brave/brave-core/assets/40611140/9f979105-2087-4901-88cb-a5caa268a4d8
